django/db/models/fields/related_descriptors.py

```diff
@@ -1130,10 +1105,10 @@ def _add_items(self, source_field_name, target_field_name, *objs):
 
                     # Add the ones that aren't there already
                     self.through._default_manager.using(db).bulk_create([
-                        self.through(**{
+                        self.through(**dict(through_defaults, **{
```

Comment:
Can replace `**dict(through_defaults, **{…})` and use unpacking generalisations: `**{**through_defaults, **{…}}`.

Reply:
I guess `**through_defaults, **{…}` would work as well given surfacing a `TypeError` if any of the ids properties are provided through `through_defaults` should probably not go unnoticed.